File: ios/Tradeguruelectrical/Services/APIConfig.swift
import Foundation

nonisolated enum APIConfig: Sendable {
    #if DEBUG
    nonisolated(unsafe) static var useMockData = false
    #endif
    static let baseURL = "https://tradeguru.com.au/api/v1"
    static let platform = "ios"
    static let appVersion = "1.0.0"
    static let workosClientId = "client_01JWQK8QD9RJVTCTMR8ACE9CKB"
    static let authProxyURL = "https://tradeguru.com.au/api/workos-auth-proxy"
    static let oauthRedirectURI = "tradeguru://auth-callback"

    static func url(_ path: String) -> URL {
        let clean = path.hasPrefix("/") ? String(path.dropFirst()) : path
        guard let url = URL(string: "\(baseURL)/\(clean)") else {
            fatalError("Invalid API URL: \(baseURL)/\(clean)")
        }
        return url
    }

    static func headers(deviceId: String, jwt: String? = nil) -> [String: String] {
        var h = [
            "Content-Type": "application/json",
            "X-Device-ID": deviceId,
        ]
        if let jwt {
            h["Authorization"] = "Bearer \(jwt)"
        }
        return h
    }

    static func request(_ path: String, method: String = "POST", deviceId: String, jwt: String? = nil) -> URLRequest {
        var req = URLRequest(url: url(path))
        req.httpMethod = method
        req.timeoutInterval = 120
        for (key, value) in headers(deviceId: deviceId, jwt: jwt) {
            req.setValue(value, forHTTPHeaderField: key)
        }
        return req
    }
}

File: android/app/src/main/java/com/tradeguru/electrical/services/APIConfig.kt
package com.tradeguru.electrical.services

import okhttp3.HttpUrl
import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
import okhttp3.Request

object APIConfig {
    const val BASE_URL = "https://tradeguru.com.au/api/v1"
    const val PLATFORM = "android"
    const val APP_VERSION = "1.0.0"
    const val WORKOS_CLIENT_ID = "client_01JWQK8QD9RJVTCTMR8ACE9CKB"
    const val AUTH_PROXY_URL = "https://tradeguru.com.au/api/workos-auth-proxy"
    const val OAUTH_REDIRECT_URI = "tradeguru://auth-callback"

    fun url(path: String): HttpUrl {
        val clean = if (path.startsWith("/")) path.drop(1) else path
        return "$BASE_URL/$clean".toHttpUrlOrNull()
            ?: BASE_URL.toHttpUrlOrNull()
            ?: throw IllegalStateException("Invalid base URL")
    }

    fun requestBuilder(
        path: String,
        method: String = "POST",
        deviceId: String,
        jwt: String? = null
    ): Request.Builder {
        val builder = Request.Builder()
            .url(url(path))
            .addHeader("Content-Type", "application/json")
            .addHeader("X-Device-ID", deviceId)

        if (jwt != null) {
            builder.addHeader("Authorization", "Bearer $jwt")
        }

        return builder
    }
}
